dm/syncer/syncer.go

```diff
@@ -2344,6 +2344,35 @@ func (s *Syncer) Run(ctx context.Context) (err error) {
 			if err2 != nil {
 				return err2
 			}
+		case *replication.TransactionPayloadEvent:
+			for _, tpev := range ev.Events {
+				switch tpevt := tpev.Event.(type) {
+				case *replication.RowsEvent:
+					eventIndex++
+					s.metricsProxies.Metrics.BinlogEventRowHistogram.Observe(float64(len(tpevt.Rows)))
+					ec.header.EventType = replication.WRITE_ROWS_EVENTv2 // FIXME
+					sourceTable, err2 = s.handleRowsEvent(tpevt, ec)
+					if sourceTable != nil && err2 == nil && s.cfg.EnableGTID {
+						if _, ok := affectedSourceTables[sourceTable.Schema]; !ok {
+							affectedSourceTables[sourceTable.Schema] = make(map[string]struct{})
+						}
+						affectedSourceTables[sourceTable.Schema][sourceTable.Name] = struct{}{}
+					}
+				case *replication.QueryEvent:
```

Comment:
seems `locationRecorder` should also handle `TransactionPayloadEvent` to correctly maintain the location property.

https://github.com/pingcap/tiflow/blob/d35ab225966b88cd6415b90a3d69b64dbcb65f54/dm/syncer/binlogstream/binlog_locations.go#L36-L55

Reply:
Something like this?
4e76bb961cc4e9b27f43d82277807055da9c7601

Or should it do something with the events embedded in the Transaction Payload Event?

Reply:
@lance6716 what do you think?

Reply:
I think https://github.com/pingcap/tiflow/commit/4e76bb961cc4e9b27f43d82277807055da9c7601 is good.

And sorry I don't have enough time to add integration test for it. we can do it later